pkg/operator/operator.go

```diff
@@ -143,42 +145,44 @@ func (c *vSphereProblemDetectorController) runChecks(ctx context.Context) (time.
 		return 0, err
 	}
 
-	var results []checkResult
 	checkContext := &check.CheckContext{
 		Context:    ctx,
 		VMConfig:   vmConfig,
 		VMClient:   vmClient,
 		KubeClient: c,
 	}
 
+	checkRunner := NewCheckThreadPool(parallelVSPhereCalls)
+	resultCollector := NewResultsCollector()
 	var errs []error
-	clusterResults, err := c.runClusterChecks(checkContext)
-	if err != nil {
+	if err := c.addClusterChecks(checkContext, checkRunner, resultCollector); err != nil {
```

Comment:
cluster check does not return any error.

Reply:
removed err